assets/js/Station.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Sign from './Sign';
+
+function zoneDescription(stationConfig, zone) {
+  if (stationConfig.zones[zone] !== true) {
+    return stationConfig.zones[zone];
+  }
+
+  const zones = {
+    n: 'NB', s: 'SB', e: 'EB', w: 'WB', m: 'MZ', c: 'CN',
+  };
+  return zones[zone];
+}
+
+function makeSign(config, zone, signs, currentTime, line) {
+  if (config.zones[zone]) {
+    const key = `${config.id}-${zone}`;
+    const lines = signs[key];
+    let lineOne;
+    let lineTwo;
+    let lineOneDuration;
+    let lineTwoDuration;
+    if (lines !== undefined) {
+      lineOne = lines[0].text;
+      lineTwo = lines[1].text;
+      lineOneDuration = lines[0].duration;
+      lineTwoDuration = lines[1].duration;
+    } else {
+      lineOne = '';
+      lineTwo = '';
+      lineOneDuration = '0';
+      lineTwoDuration = '0';
+    }
+
+    return (
+      <Sign
+        key={key}
+        signId={zoneDescription(config, zone)}
+        line={line}
+        lineOne={lineOne}
+        lineOneDuration={lineOneDuration}
+        lineTwo={lineTwo}
+        lineTwoDuration={lineTwoDuration}
+        currentTime={currentTime}
+      />
+    );
+  }
+
+  return null;
+}
+
+function Station({
+  config, signs, currentTime, line,
+}) {
+  return (
+    <div key={config.id}>
+      <h3>
+        {config.name}
+        {' '}
+        <small>
+(
+          {config.id}
+)
+        </small>
+      </h3>
+      <div className="row">
+        <div className="col">
+          {makeSign(config, 's', signs, currentTime, line)}
+          {makeSign(config, 'w', signs, currentTime, line)}
+        </div>
+        <div className="col">
+          {makeSign(config, 'c', signs, currentTime, line)}
+          {makeSign(config, 'm', signs, currentTime, line)}
+        </div>
+        <div className="col">
+          {makeSign(config, 'n', signs, currentTime, line)}
+          {makeSign(config, 'e', signs, currentTime, line)}
```

Comment:
this causes weird results for example when on the orange line, e means busway and acts as a mezz - probably ok, but i thought id mention

Reply:
Yeah, true. Maybe another reason (beyond OCHN/OCHS, etc) to change to a data structure of just specifying per-station, what signs we want where. Initially I was not a fan of putting presentational information in the configuration, but I'm coming around on not really seeing another way of doing it. But it's probably not worth doing now; rather just get this in and move on.